### PR TITLE
chore(base): setup base classes

### PR DESCRIPTION
**Pull request by MuhweziDeo**
#### Description
Setup Base classes
#### Type of change

Please select the relevant option

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Others (cosmetics, styling, improvements)
- [ ] This change requires a documentation update

#### How Has This Been Tested?

- [x] Unit
- [ ] Integration
- [ ] End-to-end

#### How to Test

Give instructions on how reviewers can verify your work

#### Any background context you want to add
Base classes will help us not to repeat ourselves 
#### Story-ID

[Finishes|Delivers|Fixes #ID]

### CODE REVIEW

File: src/base/interfaces/IController.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { Request, Response } from "express";
+
+export interface IController {
+deleteRecord: (request: Request, response: Response) => any;
+updateRecord: (request: Request, response: Response) => any;
+findAllRecords: (request: Request, response: Response) => any;
+createRecord: (request: Request, response: Response) => any;
+findOneRecord: (request: Request, reponse: Response) => any;
+}
```
Comment (fahdjamy) [inline src/base/interfaces/IController.ts:9]: @MuhweziDeo I think you might wanna indent the lines in this file sir.

File: src/base/interfaces/IService.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+export interface IService<T> {
+findOne: (options: object) => Promise<T>;
+findAll: (options?: object) => Promise<T[]>;
+deleteOne: (options: object) => Promise<boolean>;
+updateOne: (data: object, options: object) => Promise<any>;
+createOne: (data: T, options?: object) => Promise<T>;
+findOrCreate: (options: object) => Promise<any>;
```
Comment (fahdjamy) [inline src/base/interfaces/IService.ts]: The same indentation might be needed here.

**Review by fahdjamy — COMMENTED** [review]
This is some really great work @MuhweziDeo. Just left a few things you might wanna modify sir.

**Review by fahdjamy — COMMENTED** [review]
One thing though @MuhweziDeo, don't you think it might be better if we put all queries that hit the Database in a `try-catch` block or it is safer this way??

**Comment by MuhweziDeo** [comment]
> One thing though @MuhweziDeo, don't you think it might be better if we put all queries that hit the Database in a `try-catch` block or it is safer this way??

Try and catch blocks are meant to be used when you are unsure if an exception may raise but if u handle all the chances of anything going wrong then you don't need a try-catch block

**Pull request merged**